### PR TITLE
feat(design-system): improve layout showcase

### PR DESCRIPTION
**Pull request by mnavarroespinosa**
EX-7985

### CODE REVIEW

File: packages/x-tailwindcss/demo/src/css/modal.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+.modal {
+    pointer-events: none;
+    visibility: hidden;
+    position: fixed;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    opacity: 0;
+    z-index: 999;
```
Comment (ajperezbau) [inline packages/x-tailwindcss/demo/src/css/modal.css:15]: I think this `z-index` is not needed, and is causing the `copy to clipboard` message to be always hidden.

```suggestion
```

File: packages/x-tailwindcss/demo/src/components/xds-layout-fixed-header.vue [hunk 2]
```diff
@@ -1,20 +1,19 @@
 <template>
   <XdsBaseShowcase #default="{ cssClass }" title="Layout fixed header" :sections="sections">
-    <button @click="openModal(cssClass)" class="x-button">Open layout example</button>
+    <label for="layout-fixed-header-modal" class="x-button">Open layout example</label>
+    <input id="layout-fixed-header-modal" type="checkbox" class="modal-toggle" />
 
-    <dialog :ref="cssClass" class="modal">
+    <div :ref="cssClass" class="modal x-bg-neutral-0">
```
Comment (ajperezbau) [inline packages/x-tailwindcss/demo/src/components/xds-layout-fixed-header.vue:6]: I think the `ref` is not needed anymore here or in the other files.

**Review by ajperezbau — COMMENTED** [review]
Good job! 👌

File: packages/x-tailwindcss/demo/src/components/xds-layout-utils.vue [hunk 3]
```diff
@@ -97,12 +120,23 @@
...
+      Expand: 'Makes a layout item to fit the container height.'
+    };
+
     protected get sections(): ShowcaseSections {
       return {
-        'No margin': this.noMargin.map(addParentClasses(this.base)),
-        'On margin': this.onMargin,
-        Overlap: this.overlap.map(addParentClasses(this.base)),
-        Expand: this.expand.map(addParentClasses(this.base))
+        '': [this.base]
```
Comment (CachedaCodes) [inline packages/x-tailwindcss/demo/src/components/xds-layout-utils.vue:139]: Why not use `Default` here instead of `''`?

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-tailwindcss/demo/src/css/modal.css
[the remark this answers is not in the document]
Comment (mnavarroespinosa) [reply packages/x-tailwindcss/demo/src/css/modal.css:1]: Sure, good point, I've changed it!

File: packages/x-tailwindcss/demo/src/components/xds-layout-utils.vue (hunk 3 above)
Comment (mnavarroespinosa) [reply packages/x-tailwindcss/demo/src/components/xds-layout-utils.vue:138]: Using `Default` there will print the text next to the button to open the modal, I didn't see it really necessary

image

**Review by CachedaCodes — APPROVED** [review]

**Review by ajperezbau — DISMISSED** [review]
Good job 👌

**Review by ajperezbau — APPROVED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**